app/src/main/java/pl/ppiwd/exerciseanalyst/activities/TrainingDataActivity.java

```diff
@@ -77,6 +82,9 @@ private void initialize() {
         this.serverConnection = getServerConnection();
         this.deviceConnectionServiceChecker = new DeviceConnectionServiceChecker(this);
         configureButton(deviceConnectionServiceChecker.isServiceRunning());
+
+        IntentFilter filter = new IntentFilter(BroadcastMsgs.ACTIVITY_BROADCAST_INTENT_ACTION);
+        LocalBroadcastManager.getInstance(this).registerReceiver(metaWearBroadcastSink, filter);
```

Comment:
Please unregister this BraodcastReceiver in onDestroy method.  
LocalBroadcastManager.getInstance(this).unregisterReceiver(metaWearBroadcastSink);